src/index.test.ts

```diff
@@ -271,6 +289,29 @@ describe('jsonFetch', function () {
 
       throw new Error('Should have failed');
     });
+
```

Comment:
only test that is missing is one that tests what happens when a request fails in a non-retryable way, and maybe one for when we retry and it eventually does succeed (the test below only tests retries & consistent failures).